common/failure/failure.go

```diff
@@ -91,21 +102,12 @@ func Truncate(f *failurepb.Failure, maxSize int) *failurepb.Failure {
 		}}
 	}
 
-	if len(f.Message) > maxSize {
-		newFailure.Message = f.Message[:maxSize]
-		return newFailure
+	newFailure.Source = trunc(f.Source)
+	newFailure.Message = trunc(f.Message)
+	newFailure.StackTrace = trunc(f.StackTrace)
+	if f.Cause != nil && maxSize > 0 && maxDepth > 0 {
```

Comment:
is it possible that we have like one byte left but none of the fields can be truncated and we just keep creating empty nested failures.

Reply:
It could. I added maxDepth so at least it'll bottom out somewhere, but the empty nested ones would look weird.

We could change this to only recurse if maxDepth is > 20 and also subtract 10 from it each level so we get at least 10 bytes of something?

Reply:
yeah that could work. Or just check if maxSize changed or not in current level and don't recurse if not?

Reply:
Check the new version, it shouldn't leave empty causes